packages/fxa-auth-server/lib/routes/emails.js

```diff
@@ -789,58 +729,41 @@ module.exports = (
 
         log.begin('Account.RecoveryEmailSetPrimary', request);
 
-        return customs
-          .check(request, primaryEmail, 'setPrimaryEmail')
-          .then(() => {
-            if (sessionToken.tokenVerificationId) {
-              throw error.unverifiedSession();
-            }
-          })
-          .then(setPrimaryEmail)
-          .then(() => {
-            return {};
-          });
+        await customs.check(request, primaryEmail, 'setPrimaryEmail');
 
-        function setPrimaryEmail() {
-          return db
-            .getSecondaryEmail(email)
-            .then(email => {
-              if (email.uid !== uid) {
-                throw error.cannotChangeEmailToUnownedEmail();
-              }
-
-              if (!email.isVerified) {
-                throw error.cannotChangeEmailToUnverifiedEmail();
-              }
+        if (sessionToken.tokenVerificationId) {
+          throw error.unverifiedSession();
+        }
 
-              if (email.isPrimary) {
-                return;
-              }
+        const secondaryEmail = await db.getSecondaryEmail(email);
+        if (secondaryEmail.uid !== uid) {
+          throw error.cannotChangeEmailToUnownedEmail();
+        }
 
-              return db.setPrimaryEmail(uid, email.normalizedEmail);
-            })
-            .then(() => {
-              request.app.devices.then(devices =>
-                push.notifyProfileUpdated(uid, devices)
-              );
-              log.notifyAttachedServices('primaryEmailChanged', request, {
-                uid,
-                email: email,
-              });
+        if (!secondaryEmail.isVerified) {
+          throw error.cannotChangeEmailToUnverifiedEmail();
+        }
 
-              return db.account(uid);
-            })
-            .then(account => {
-              return mailer.sendPostChangePrimaryEmail(
-                account.emails,
-                account,
-                {
-                  acceptLanguage: request.app.acceptLanguage,
-                  uid,
-                }
-              );
-            });
+        if (!secondaryEmail.isPrimary) {
```

Comment:
What happens if `secondaryEmail.isPrimary` is already set? Do you know how that'd happen? Should we only do any of the notification/mailing if `isPrimary` is false, otherwise we'd be notifying for an address that's already marked primary.

Reply:
Yes, good catch.  Thanks!